From a8f51f47ceda5bfac14d1c45b3c7c214efb89561 Mon Sep 17 00:00:00 2001
From: Boon Khai Ng <boon.khai.ng@intel.com>
Date: Mon, 30 Aug 2021 16:23:02 +0800
Subject: [PATCH 42/42] HSD #18017770485: intel: fcs: Update Certificate Mask
 for FPGA Attestation Update the certificate mask to 0xff to cover all
 attestation certificate in Agilex Family

commit d68144779f2b613cbd1eb892b6e06aeffff54009 from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Signed-off-by: Boon Khai Ng <boon.khai.ng@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/crypto/intel_fcs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/intel_fcs.c b/drivers/crypto/intel_fcs.c
index d9bb528d2c82..7ea594329fae 100644
--- a/drivers/crypto/intel_fcs.c
+++ b/drivers/crypto/intel_fcs.c
@@ -983,7 +983,7 @@ static long fcs_ioctl(struct file *file, unsigned int cmd,
 		msg->payload_length = 0;
 		msg->payload_output = d_buf;
 		msg->payload_length_output = CERTIFICATE_RSP_MAX_SZ;
-		msg->arg[0] = data->com_paras.certificate.c_request & 0x001f;
+		msg->arg[0] = data->com_paras.certificate.c_request & 0x00ff;
 		priv->client.receive_cb = fcs_attestation_callback;
 
 		ret = fcs_request_service(priv, (void *)msg,
@@ -1022,7 +1022,7 @@ static long fcs_ioctl(struct file *file, unsigned int cmd,
 		}
 
 		msg->command = COMMAND_FCS_ATTESTATION_CERTIFICATE_RELOAD;
-		msg->arg[0] = data->com_paras.c_reload.c_request & 0x001f;
+		msg->arg[0] = data->com_paras.c_reload.c_request & 0x00ff;
 		priv->client.receive_cb = fcs_vab_callback;
 		ret = fcs_request_service(priv, (void *)msg,
 					  10 * FCS_REQUEST_TIMEOUT);
-- 
2.31.1

